crates/red_knot_python_semantic/src/suppression.rs

```diff
@@ -1,50 +1,100 @@
-use salsa;
+use std::cmp::Ordering;
 
-use ruff_db::{files::File, parsed::comment_ranges, source::source_text};
+use ruff_python_parser::TokenKind;
+use ruff_source_file::LineRanges;
+use ruff_text_size::{Ranged, TextRange, TextSize};
+
+use ruff_db::{files::File, parsed::parsed_module, source::source_text};
 use ruff_index::{newtype_index, IndexVec};
 
 use crate::{lint::LintId, Db};
 
 #[salsa::tracked(return_ref)]
-pub(crate) fn suppressions(db: &dyn Db, file: File) -> IndexVec<SuppressionIndex, Suppression> {
```

Comment:
Whoops. I accidentally commited this file when I rebased https://github.com/astral-sh/ruff/pull/14956/files#diff-b1807b646317ac1945d748f7db40451ac62c582b1bbc049b174e8d98f13d3f22 Probably because it didn't get stashed with `git stash`. So consider all code in this file as new